tensorrt_llm/_torch/modules/fused_moe.py

```diff
@@ -817,19 +817,27 @@ def forward_chunk(
 
         if self.use_dp and self.parallel_size > 1 and not disable_fp4_allgather(
         ) and not self.enable_alltoall:
-            if x_sf is None:
-                x, token_selected_experts, token_final_scales = allgather(
-                    [x, token_selected_experts, token_final_scales],
-                    self.mapping,
-                    dim=0,
-                    sizes=None if use_dp_padding else all_rank_num_tokens)
-            else:
-                # Fp4 gemm has extra scaling factor
-                x, x_sf, token_selected_experts, token_final_scales = allgather(
-                    [x, x_sf, token_selected_experts, token_final_scales],
-                    self.mapping,
-                    dim=0,
-                    sizes=None if use_dp_padding else all_rank_num_tokens)
+            var_dict = {
+                'x': x,
+                'x_sf': x_sf,
+                'token_selected_experts': token_selected_experts,
+                'token_final_scales': token_final_scales,
+            }
+            key_list = [key for key, val in var_dict.items() if val is not None]
```

Comment:
Can this use the builtin `filter()` method, the line below can also be done with a `map()`